src/nodes.coffee

```diff
@@ -3298,17 +3304,24 @@ exports.Op = class Op extends Base
   constructor: (op, first, second, flip, {@invertOperator, @originalOperator = op} = {}) ->
     super()
 
-    if op is 'do'
-      return Op::generateDo first
     if op is 'new'
-      if (firstCall = first.unwrap()) instanceof Call and not firstCall.do and not firstCall.isNew
-        return firstCall.newInstance()
-      first = new Parens firstCall if firstCall instanceof Code and firstCall.bound or firstCall.do
+      if ((firstCall = unwrapped = first.unwrap()) instanceof Call or (firstCall = unwrapped.base) instanceof Call) and not firstCall.do and not firstCall.isNew
+        return new Value firstCall.newInstance(), if firstCall is unwrapped then [] else unwrapped.properties
+      first = new Parens first unless first instanceof Parens or first.unwrap() instanceof IdentifierLiteral or first.hasProperties?()
```

Comment:
Compared to the previous version
```
first = new Parens firstCall if firstCall instanceof Code and firstCall.bound or firstCall.do
```
, this takes a more conservative approach and defaults to parenthesizing the argument to "simple" (no-arg/no-call-parens) `new` unless it's clearly something that doesn't need to be parenthesized in the output eg `new A` or `new A.B`

Added a test case checking that a more complex argument (`new try A`) parenthesizes its output correctly; this test would have failed with the new "always convert `new` to a `Call`" approach without this